### ISSUE TITLE
Package Fails to Precompile

### ISSUE THREAD

**Issue by frawas** [QuantEcon/QuantEcon.jl#251]
Looks like some dependencies were changed, so QuantEcon and Games.jl stopped working for me (even after Pkg.rm() and re-installing the packages)

Error:
julia> using QuantEcon
[ Info: Precompiling QuantEcon [fcd29c91-0bd7-5a09-975d-7ac3f643a60c]
WARNING: both FFTW and Util export "Frequencies"; uses of it in module Periodograms must be qualified
ERROR: LoadError: LoadError: UndefVarError: Frequencies not defined

How do I get it to work again?

**Comment by Shunsuke-Hori** [comment]
@frawas I cannot reproduce the error. Would you try `update` and `precompile` again?

**Comment by frawas** [comment]
Still nothing! I'm unfamiliar with this error or what is causing it, I have been using the package as well as Games.jl fine.

julia> Pkg.update("QuantEcon")
  Updating registry at `C:\Users\Farah\.julia\registries\General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
 Resolving package versions...
 Installed TimerOutputs ─ v0.5.2
 Installed HTTP ───────── v0.8.8
  Updating `C:\Users\Farah\.julia\environments\v1.1\Project.toml`
 [no changes]
  Updating `C:\Users\Farah\.julia\environments\v1.1\Manifest.toml`
  [cd3eb016] ↑ HTTP v0.8.7 ⇒ v0.8.8
  [a759f4b9] ↑ TimerOutputs v0.5.1 ⇒ v0.5.2

julia> using QuantEcon
[ Info: Precompiling QuantEcon [fcd29c91-0bd7-5a09-975d-7ac3f643a60c]
WARNING: both FFTW and Util export "Frequencies"; uses of it in module Periodograms must be qualified
ERROR: LoadError: LoadError: UndefVarError: Frequencies not defined
Stacktrace:
 [1] top-level scope at none:0
 [2] include at .\boot.jl:326 [inlined]
 [3] include_relative(::Module, ::String) at .\loading.jl:1038
 [4] include at .\sysimg.jl:29 [inlined]
 [5] include(::String) at C:\Users\Farah\.julia\packages\DSP\wwKNu\src\DSP.jl:1
 [6] top-level scope at none:0
 [7] include at .\boot.jl:326 [inlined]
 [8] include_relative(::Module, ::String) at .\loading.jl:1038
 [9] include(::Module, ::String) at .\sysimg.jl:29
 [10] top-level scope at none:2
 [11] eval at .\boot.jl:328 [inlined]
 [12] eval(::Expr) at .\client.jl:404
 [13] top-level scope at .\none:3
in expression starting at C:\Users\Farah\.julia\packages\DSP\wwKNu\src\periodograms.jl:186
in expression starting at C:\Users\Farah\.julia\packages\DSP\wwKNu\src\DSP.jl:13
ERROR: LoadError: Failed to precompile DSP [717857b8-e6f2-59f4-9121-6e50c889abd2] to C:\Users\Farah\.julia\compiled\v1.1\DSP\OtML7.ji.
Stacktrace:
 [1] error(::String) at .\error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at .\loading.jl:1197
 [3] _require(::Base.PkgId) at .\loading.jl:960
 [4] require(::Base.PkgId) at .\loading.jl:858
 [5] require(::Module, ::Symbol) at .\loading.jl:853
 [6] include at .\boot.jl:326 [inlined]
 [7] include_relative(::Module, ::String) at .\loading.jl:1038
 [8] include(::Module, ::String) at .\sysimg.jl:29
 [9] top-level scope at none:2
 [10] eval at .\boot.jl:328 [inlined]
 [11] eval(::Expr) at .\client.jl:404
 [12] top-level scope at .\none:3
in expression starting at C:\Users\Farah\.julia\packages\QuantEcon\vVeHL\src\QuantEcon.jl:12
ERROR: Failed to precompile QuantEcon [fcd29c91-0bd7-5a09-975d-7ac3f643a60c] to C:\Users\Farah\.julia\compiled\v1.1\QuantEcon\V0Mv9.ji.
Stacktrace:
 [1] compilecache(::Base.PkgId, ::String) at .\loading.jl:1197
 [2] _require(::Base.PkgId) at .\loading.jl:960
 [3] require(::Base.PkgId) at .\loading.jl:858
 [4] require(::Module, ::Symbol) at .\loading.jl:853

**Comment by Nosferican** [comment]
I can confirm using it with  project environment works fine. This is likely an issue of having a polluted environment. The recommended way to use Julia is with project environments which avoid those issues.

**Comment by Nosferican** [comment]
For those that have the issue, I would suggest using project environments (see [Working with environments](https://julialang.github.io/Pkg.jl/v1/environments/)).

Your global environment file should probably only include packages for cross projects, such as IDE related packages (e.g., Atom, Juno, IJulia), package development (e.g., Coverage, Documenter, DocumenterTools), and tooling (e.g., BenchmarkTools, Coverage, Revise, Weave).
```
(v1.3) pkg> st
    Status `~/.julia/environments/v1.3/Project.toml`
  [c52e3926] Atom v0.11.3
  [6e4b80f9] BenchmarkTools v0.4.3
  [a2441757] Coverage v0.9.3
  [e30172f5] Documenter v0.23.4
  [35a29f4d] DocumenterTools v0.1.2
  [7073ff75] IJulia v1.20.2
  [e5e0dc1b] Juno v0.7.2
  [295af30f] Revise v2.3.0
  [40c74d1a] TableView v0.4.1
  [44d3d7a6] Weave v0.9.1
  [0f1e0344] WebIO v0.8.11
```
Those would be available for every environment. Other packages should be added to the project environment.
```
(v1.3) pkg> activate .
Activating new environment at `Path/to/my/ProjectName/Project.toml`

(ProjectName) pkg> st
    Status `Path/to/my/ProjectName/Project.toml`
  (no changes since last commit)

(ProjectName) pkg> add QuantEcon
  Updating registry at `~/.julia/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
 Resolving package versions...
  Updating `Path/to/my/ProjectName/Project.toml`
  [fcd29c91] + QuantEcon v0.16.2
  Updating `Path/to/my/ProjectName/Manifest.toml`
  [621f4979] + AbstractFFTs v0.5.0
  [ec485272] + ArnoldiMethod v0.0.4
  [7d9fca2a] + Arpack v0.3.1
  [4fba245c] + ArrayInterface v1.2.1
  [9e28174c] + BinDeps v0.8.10
  [b99e7846] + BinaryProvider v0.5.8
  [631607c0] + CMake v1.1.2
  [d5fb7624] + CMakeWrapper v0.2.3
  [49dc2e85] + Calculus v0.5.0
  [bbf7d656] + CommonSubexpressions v0.2.0
  [34da2185] + Compat v2.2.0
  [8f4d0f93] + Conda v1.3.0
  [717857b8] + DSP v0.6.1
  [9a962f9c] + DataAPI v1.1.0
  [864edb3b] + DataStructures v0.17.5
  [01453d9d] + DiffEqDiffTools v1.4.0
  [163ba53b] + DiffResults v0.0.4
  [b552c78f] + DiffRules v0.1.0
  [31c24e10] + Distributions v0.21.8
  [7a1cc6ca] + FFTW v1.1.0
  [1a297f60] + FillArrays v0.7.4
  [f6369f11] + ForwardDiff v0.10.6
  [d25df0c9] + Inflate v0.1.1
  [682c06a0] + JSON v0.21.0
  [093fc24a] + LightGraphs v1.3.0
  [d3d80556] + LineSearches v7.0.1
  [1914dd2f] + MacroTools v0.5.2
  [fdba3010] + MathProgBase v0.7.7
  [e1d29d7a] + Missings v0.4.3
  [d41bc354] + NLSolversBase v7.5.0
  [76087f3c] + NLopt v0.5.1
  [77ba4419] + NaNMath v0.3.2
  [429524aa] + Optim v0.19.4
  [bac558e1] + OrderedCollections v1.1.0
  [90014a1f] + PDMats v0.9.10
  [d96e819e] + Parameters v0.12.0
  [69de0a69] + Parsers v0.3.8
  [f27b6e38] + Polynomials v0.6.0
  [85a6dd25] + PositiveFactorizations v0.2.3
  [27ebfcd6] + Primes v0.4.0
  [1fd47b50] + QuadGK v2.1.1
  [fcd29c91] + QuantEcon v0.16.2
  [3cdcf5f2] + RecipesBase v0.7.0
  [189a3867] + Reexport v0.2.0
  [ae029012] + Requires v0.5.2
  [79098fc4] + Rmath v0.5.1
  [699a6c99] + SimpleTraits v0.9.1
  [a2af1166] + SortingAlgorithms v0.3.1
  [276daf66] + SpecialFunctions v0.8.0
  [90137ffa] + StaticArrays v0.12.1
  [2913bbd2] + StatsBase v0.32.0
  [4c63d2b9] + StatsFuns v0.9.0
  [30578b45] + URIParser v0.4.0
  [81def892] + VersionParsing v1.1.3
  [2a0f44e3] + Base64 
  [ade2ca70] + Dates 
  [8bb1440f] + DelimitedFiles 
  [8ba89e20] + Distributed 
  [b77e0a4c] + InteractiveUtils 
  [76f85450] + LibGit2 
  [8f399da3] + Libdl 
  [37e2e46d] + LinearAlgebra 
  [56ddb016] + Logging 
  [d6f4376e] + Markdown 
  [a63ad114] + Mmap 
  [44cfe95a] + Pkg 
  [de0858da] + Printf 
  [3fa0cd96] + REPL 
  [9a3f8284] + Random 
  [ea8e919c] + SHA 
  [9e88b42a] + Serialization 
  [1a1011a3] + SharedArrays 
  [6462fe0b] + Sockets 
  [2f01184e] + SparseArrays 
  [10745b16] + Statistics 
  [4607b0f0] + SuiteSparse 
  [8dfed614] + Test 
  [cf7118a7] + UUIDs 
  [4ec0a83e] + Unicode 

(ProjectName) pkg> add https://github.com/QuantEcon/Games.jl
  Updating git-repo `https://github.com/QuantEcon/Games.jl`
 Resolving package versions...
  Updating `Path/to/my/ProjectName/Project.toml`
  [d5cce6e7] + Games v0.0.0 #master (https://github.com/QuantEcon/Games.jl)
  Updating `Path/to/my/ProjectName/Manifest.toml`
  [6e4b80f9] + BenchmarkTools v0.4.3
  [e2554f3b] + Clp v0.7.0
  [3da002f7] + ColorTypes v0.8.0
  [861a8166] + Combinatorics v0.7.0
  [53c48c17] + FixedPointNumbers v0.6.1
  [d5cce6e7] + Games v0.0.0 #master (https://github.com/QuantEcon/Games.jl)
  [4d00f742] + GeometryTypes v0.7.6
  [c8e1da08] + IterTools v1.2.0
  [4076af6c] + JuMP v0.20.1
  [b8f27783] + MathOptInterface v0.9.7
  [774612a8] + ParameterJuMP v0.1.2
  [67491407] + Polyhedra v0.5.6

julia> using QuantEcon, Games
[ Info: Precompiling QuantEcon [fcd29c91-0bd7-5a09-975d-7ac3f643a60c]
[ Info: Precompiling Games [d5cce6e7-4428-536b-a17f-94dfa0e3a07a]
```

**Comment by galenlynch** [comment]
This is probably fixed by JuliaRegistries/General#5778
